glusterd2/commands/volumes/volume-edit.go

```diff
@@ -51,6 +51,12 @@ func volumeEditHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	metadataSize := volinfo.GetMetadataSize()
```

Comment:
This doesn't handle replacing the keys case.

```go
if req.MetadataSize() > maxMetadataSizeLimit {
    // return error
}

// update volinfo.Metadata and replace keys, as done below

if volinfo.MetadataSize() > maxMetadataSizeLimit {
    return error
}

```

Use the same logic for peer edit

Reply:
Ooh yes, you are right. So, sorry. Will update it.